cloudinit/cmd/status.py

```diff
@@ -93,6 +93,36 @@ class StatusDetails(NamedTuple):
 {description}"""
 
 
+def query_systemctl(
```

Comment:
This is a good utility function to generalize interaction with systemctl. While we see other call-sites throughout cloud-init to systemctl in cloudini/net/network_manager, cloudinit/net/activators.py, cloudinit/sources/helpers/azure.py and cloudinit/config/cc_mounts.py, these entry points are much later in boot than a potential early-boot invocation of cloud-init status from external consumers. So I don't necessarily think we "need" this logic elsewhere for this PR. It may be worth generalizing it in util or somewhere if we find another call-site that warrants early boot interaction and exposure to the socket not yet being responsive.

Reply:
The current implemntation is very status specific as well. So I don't think it's worth generalizing further at this time.